cloudmarker/test/test_esstore.py

```diff
@@ -0,0 +1,73 @@
+"""Tests for EsStore plugin."""
+
+import unittest
+from unittest import mock
+
+from elasticsearch import ElasticsearchException
+
+from cloudmarker.stores import esstore
+
+
+class EsStoreTest(unittest.TestCase):
+    """Tests for EsStore plugin."""
+
+    @mock.patch('cloudmarker.stores.esstore.Elasticsearch.bulk')
+    def test_happy_flow(self, mock_bulk):
+        mock_record = {'record_type': 'firewall_rule'}
+
+        es_store = esstore.EsStore()
+
+        mock_bulk_response = {
+            'errors': '',
+            'items': [
+                {'index': {'status': 200}}
+            ]
+        }
+
+        mock_bulk.return_value = mock_bulk_response
+
+        es_store.write(mock_record)
+        es_store.done()
+
+        mock_bulk.assert_called_once()
+        # In case of successful push the buffer should reset.
+        self.assertEqual(es_store._buffer, '')
+
+    @mock.patch('cloudmarker.stores.esstore.Elasticsearch.bulk')
+    def test_exception_flow(self, mock_bulk):
+        mock_record = {'record_type': 'firewall_rule'}
+
+        es_store = esstore.EsStore()
+
+        mock_bulk.side_effect = ElasticsearchException()
+
+        es_store.write(mock_record)
+        es_store.done()
+
+        # In case of any expception the buffer should not reset.
+        self.assertNotEqual(es_store._buffer, '')
+
+    @mock.patch('cloudmarker.stores.esstore.Elasticsearch.bulk')
+    def test_bulk_response_error(self, mock_bulk):
+        mock_record = {'record_type': 'firewall_rule'}
+
+        es_store = esstore.EsStore()
+
+        mock_bulk_response = {
+            'errors': 'foo',
```

Comment:
Also forgot to mention, the `errors` key here is supposed to have a value of `true` or `false` though it doesn't affect or break the test in any way and having the value to be `foo` is also considered as `true`.